### ISSUE TITLE
Is there more experiment between End2End and other lane detection methods?

### ISSUE THREAD

**Issue by hlnull** [wvangansbeke/LaneDetection_End2End#3]
Hi, @wvangansbeke  

Thanks for your great work. The result seems impressive.
Well, I am wondering if you compare the End2End method with other lane detection methods, https://arxiv.org/abs/1712.06080 and https://arxiv.org/abs/1802.05591?

**Comment by wvangansbeke** [comment]
Hi @hlnull 
I'm considering to carry out more experiments, but I'm also working on other projects at the moment. I find our method elegant to make lane detection end-to-end. It seems logical to me that if you can optimize towards the final line coordinates directly, better results are expected. 

Kind regards,
Wouter.

**Comment by wvangansbeke** [comment]
Hi,
The validation of the model is performed automatically (20% of the dataset). I didn't include the test code here since I validate using this validation set. 

Kind regards.
Wouter

**Comment by yinhai86924** [comment]
how to test your model , is there a test code?  I only see the training code, thank you !

**Comment by yinhai86924** [comment]
I have some trouble about how to generate W use deep learning?

**Comment by wvangansbeke** [comment]
Hi @yinhai86924 
I use a grid with X, Y coordinates which is transformed to a bird's eye view. In this way the features (=weights) can be extracted in the normal view. The network learns these weights in an end-to-end fashion by using the ground truth (=lane line coefficients).

Can you maybe link to a specific line in the code if something is not clear? 

Kind regards.

**Comment by yinhai86924** [comment]
> Hi @yinhai86924
> I use a grid with X, Y coordinates which is transformed to a bird's eye view. In this way the features (=weights) can be extracted in the normal view. The network learns these weights in an end-to-end fashion by using the ground truth (=lane line coefficients).
> 
> Can you maybe link to a specific line in the code if something is not clear?
> 
> Kind regards.



> Hi @yinhai86924
> I use a grid with X, Y coordinates which is transformed to a bird's eye view. In this way the features (=weights) can be extracted in the normal view. The network learns these weights in an end-to-end fashion by using the ground truth (=lane line coefficients).
> 
> Can you maybe link to a specific line in the code if something is not clear?
> 
> Kind regards.

Does it mean that the weight value is under the normal view, the coordinate value needs to be converted to the top view for training, whether the ground truth should also be converted to the top view so as to training?

**Comment by wvangansbeke** [comment]
Hi @yinhai86924,

I think you got it. The weight maps are indeed in the normal view and the ground values in the bird's eye view.

Kind regards
Wouter
